src/main/java/core/basesyntax/service/FruitShopService.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.service;
+
+import core.basesyntax.exception.UnknownOperationException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.OperationType;
+import core.basesyntax.strategy.BalanceOperationHandler;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.PurchaseOperationHandler;
+import core.basesyntax.strategy.ReturnOperationHandler;
+import core.basesyntax.strategy.SupplyOperationHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FruitShopService {
+    private Map<String, Integer> fruitInventory = new HashMap<>();
+    private Map<OperationType, OperationHandler> operationStrategy = new HashMap<>();
+
+    public FruitShopService(Map<OperationType, OperationHandler> operationStrategy) {
+        this.operationStrategy.put(OperationType.BALANCE, new BalanceOperationHandler());
+        this.operationStrategy.put(OperationType.SUPPLY, new SupplyOperationHandler());
+        this.operationStrategy.put(OperationType.PURCHASE, new PurchaseOperationHandler());
+        this.operationStrategy.put(OperationType.RETURN, new ReturnOperationHandler());
+    }
+
+    public void processTransactions(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationType operation = transaction.getOperation();
+            String fruit = transaction.getFruit();
+            int quantity = transaction.getQuantity();
+            OperationHandler handler = operationStrategy.get(operation);
+            if (handler == null) {
+                throw new UnknownOperationException("Unknown operation type: " + operation);
+            }
+            handler.apply(fruitInventory, fruit, quantity);
+        }
+    }
```

Comment:
The `processTransactions` method is directly manipulating the `fruitInventory` map. This logic should be encapsulated within the operation handlers to adhere to the Single Responsibility Principle. Each handler should be responsible for updating the inventory based on the transaction it processes.